chain/ethereum.ts

```diff
@@ -352,6 +352,48 @@ export namespace ethereum {
     size: BigInt | null
   }
 
+  /**
+   * An Ethereum block with transactions.
+   */
+  export class FullBlock {
```

Comment:
Why not less boilerplate with:

```typescript
export class FullBlock extends Block {
   transactions: Array<Transaction>
}
```

The same trick applies to the other classes that have been introduced. I'm a fan of using interfaces for consistency personally, but don't feel the urge to evangelize it too much today. Here's what that would look like (if I remember correctly, my TS is rusty... lol):
```typescript
export type FullBlock = Block & {
   transactions: Array<Transaction>,
}
```
One nice thing about interfaces here is that if `TransactionWithReceipt` implements `Transaction` then `FullBlockWithReceipts` also implements `FullBlock`. Type/Interface compatibility isn't a huge concern for this use-case, but it's interesting to discuss.